### PR TITLE
IEN 65 Manually add or edit recruitment status and Applicant Job info

### PR DESCRIPTION
**Pull request by bhumin-fw**
Added new API for Add/Edit Applicant Job and recruitment status.
Added 2 new master tables/Apis to support Applicant job modal.

### CODE REVIEW

File: apps/api/src/applicant/ienapplicant.service.ts [hunk 1]
```diff
@@ -155,80 +173,156 @@ export class IENApplicantService {
...
-    await this.ienapplicantRepository.update(applicant.id, {
-      ...dataToUpdate,
-    });
+    dataToUpdate.end_date = null;
+    if (end_date) {
+      dataToUpdate.start_date = end_date;
+    }
 
-    const applicant_obj = await this.getApplicantById(id);
+    let job = null;
```
Comment (dbayly-freshworks) [inline apps/api/src/applicant/ienapplicant.service.ts:203]: would `let job;` work here?

**Review by dbayly-freshworks — APPROVED** [review]
Looks good over all

File: packages/common/src/dto/ienapplicant-add-status.dto.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { IsDateString, IsOptional, IsString } from 'class-validator';
+
+export class IENApplicantAddStatusDTO {
+  @IsString()
+  status!: string;
+
+  @IsString()
+  @IsOptional()
+  job_id?: string;
```
[the remark this answers is not in the document]
Comment (bhumin-fw) [reply packages/common/src/dto/ienapplicant-add-status.dto.ts:9]: We are checking it for Job objects only, and not storing it, So length won't make any impact here. But let me check for a field where we do store varchar.

File: apps/api/src/applicant/ienapplicant.service.ts [hunk 3]
```diff
@@ -1,17 +1,20 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
```
[the remark this answers is not in the document]
Comment (bhumin-fw) [reply apps/api/src/applicant/ienapplicant.service.ts:1]: agree

File: apps/api/src/applicant/ienapplicant.service.ts (hunk 1 above)
Comment (bhumin-fw) [reply apps/api/src/applicant/ienapplicant.service.ts:203]: Yes, the job would be null if other than recruitment milestone add/update.

**Pull request merged**